backend/src/main/java/com/google/blackswan/mock/SimpleRelatedDataGenerator.java

```diff
@@ -0,0 +1,120 @@
+public class SimpleRelatedDataGenerator implements RelatedDataGenerator {
+  private static final String TARGET_DIMENSION = "Ramen";
+  private static final String RELATED_METRIC = "Interest Over Time";
+  private static final String CONFIG_USERNAME = "catyu@";
+  private static final ImmutableMap<String, String> FILE_LOCATIONS = ImmutableMap.of(
+                                                                      "Udon", "/udon-data.csv",
+                                                                      "Pho", "/pho-data.csv"
+                                                                    );
+
+  private static final SimpleRelatedDataGenerator INSTANCE = new SimpleRelatedDataGenerator();
+
+  private Multimap<String, DataInfo> relatedDataMap;
+  private Map<String, Map<Timestamp, Integer>> csvDataCache;
+
+  private SimpleRelatedDataGenerator() {
+    relatedDataMap = ArrayListMultimap.create();
+    csvDataCache = new HashMap<>();
+    prefillRelatedData(); // TODO: Remove when connected to datastore.
+  }
+
+  /** 
+   * Simulates action of querying configs from datastore and adding to 
+   * relatedDataMap. Currently, manually adds these relationships.
+   */
+  private void prefillRelatedData() {
+    // TODO: Replace with call for querying configs from datastore.
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Udon", CONFIG_USERNAME));
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Pho", CONFIG_USERNAME));
+    // Multimap looks like this right now: {{Ramen,{Udon, Pho}}, ...}.
+  }
+
+  public List<RelatedData> getRelatedData(String metricName, String dimensionName, 
+      Timestamp startTime, Timestamp endTime) {
+    List<RelatedData> relatedDataList = new ArrayList<RelatedData>();
+    // TODO: Currently only tries to find related data for dimension, ex. Ramen -> Pho, 
+    //       does not include logic for metric, should be similar.
+    if (relatedDataMap.containsKey(dimensionName)) {
+      for (DataInfo relatedTopic : relatedDataMap.get(dimensionName)) {
+        Map<Timestamp, MetricValue> dataPointsPlot = 
+            getDataPointsInRange(relatedTopic.getDimensionName(), startTime, endTime);
+        
+        if (dataPointsPlot.isEmpty()) {
+          continue;
+        }
+
+        relatedDataList.add(new RelatedData(
+            relatedTopic.getUsername(), relatedTopic.getMetricName(), 
+            relatedTopic.getDimensionName(), dataPointsPlot)
+          );        
+      }
+    }
+    // When neither dimension name nor metric name is found in relatedDataMap, empty
+    // list is returned. 
+    return relatedDataList;
+  }
+
+  private ImmutableMap<Timestamp, Integer> getTopicDataPoints(String topic) {
+    if (!csvDataCache.containsKey(topic)) {
+      csvDataCache.put(topic, CSVParser.parseCSV(
+        SimpleRelatedDataGenerator.class.getResourceAsStream(FILE_LOCATIONS.get(topic))
+      ));
+    }
+
+    return ImmutableMap.copyOf(csvDataCache.get(topic));
+  }
```

Comment:
return csvDataCache.computeIfAbsent(topic, k -> CSVParser.parseCSV(
        SimpleRelatedDataGenerator.class.getResourceAsStream(FILE_LOCATIONS.get(k)))

Reply:
https://docs.google.com/document/d/1aQJEihr8XMje82aBjmmEuCImibkEGWZdiSdaUkiU2ek/edit?hl=en#heading=h.ex9aa8qaybch for some of these `computeIfAbsent` type fn's in java 8 :)

Reply:
thanks for sharing the doc! I really like concise functions hahah